src/biogeochem/DUSTMod.F90

```diff
@@ -924,4 +1219,67 @@ subroutine InitDustVars(this, bounds)
 
   end subroutine InitDustVars
 
+  !==============================================================================
+  subroutine CalcDragPartition(this, bounds, prigent_roughness_stream )
+    !
+    ! !DESCRIPTION:
+    ! Commented below by Danny M. Leung 31 Dec 2022
+    ! Calculate the drag partition effect of friction velocity due to surface roughness following
+    ! Leung et al. (2022).  This module is used in the dust emission module DUSTMod.F90 for
+    ! calculating drag partitioning. The drag partition equation comes from Marticorena and
+    ! Bergametti (1995) with constants modified by Darmenova et al. (2009). Here it is assumed
+    ! that this equation is used only over arid/desertic regions, such that Catherine Prigent's
+    ! roughness measurements represents mostly rocks. For more vegetated areas, the vegetation
+    ! roughness and drag partitioning are calculated in the DustEmission subroutine. This
+    ! subroutine is used in the InitCold subroutine of DUSTMod.F90.
+    !
+    ! !USES:
+    use PatchType               , only : patch
+    use landunit_varcon         , only : istdlak
+    use LandunitType            , only : lun
+    !
+    ! !ARGUMENTS:
+    implicit none
+    class(dust_type) , intent(inout) :: this
+    type(bounds_type), intent(in) :: bounds
+    type(prigentroughnessstream_type), intent(in) :: prigent_roughness_stream
+    !
+    ! !LOCAL VARIABLES:
+    integer  :: g, p, fp, l    ! Indices
+    real(r8) :: z0s         ! smooth roughness length (m)
+
+    ! constants
+    real(r8), parameter :: D_p = 130e-6_r8           ! [m] Medium soil particle diameter, assuming a global constant
+                                                     ! of ~130 um following Leung et al. (2022)
+    real(r8), parameter :: X = 10_r8                 ! [m] distance downwind of the roughness element (rock). Assume
+                                                     ! estimating roughness effect at a distance of 10 m following Leung et al. (2022)
+    character(len=*), parameter :: subname = 'PrigentRoughnessStream::CalcDragPartition'
+    !---------------------------------------------------------------------
+
+    ! Make sure we've been initialized
+    if ( .not. prigent_roughness_stream%IsStreamInit() )then
+       call endrun(msg=subname//' ERROR Streams have not been initialized, make sure Init is called first' &
+                              //', and streams are on')
+    end if
+
+    ! dmleung: this loop calculates the drag partition effect (or roughness effect) of rocks.
+    !          We save the drag partition factor as a patch level quantity.
+    ! TODO: EBK 02/13/2024: Several magic numbers here that should become parameters so the meaning is preserved
+    z0s = 2_r8 * D_p / 30_r8 ! equation from Frank M. White (2006).
```

Comment:
Some magic numbers here and in the following several lines. I am changing them into parameters.

Reply:
@dmleung completed this.